libp2p/network/network.go

```diff
@@ -0,0 +1,273 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	libp2pHost "github.com/libp2p/go-libp2p/core/host"
+
+	"github.com/pokt-network/pocket/libp2p/protocol"
+	"github.com/pokt-network/pocket/p2p/providers"
+	"github.com/pokt-network/pocket/p2p/providers/addrbook_provider"
+	persABP "github.com/pokt-network/pocket/p2p/providers/addrbook_provider/persistence"
+	"github.com/pokt-network/pocket/p2p/providers/current_height_provider"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+type libp2pNetwork struct {
+	base_modules.IntegratableModule
+
+	logger *modules.Logger
+	//nolint:unused // bus is used by embedded base module(s)
+	bus         modules.Bus
+	host        libp2pHost.Host
+	topic       *pubsub.Topic
+	addrBookMap typesP2P.AddrBookMap
+}
+
+const (
+	year = time.Hour * 24 * 365
+	// TECHDEBT: consider more carefully and parameterize.
+	defaultPeerTTL = 2 * year
```

Comment:
Not actionable: Wow 2 years of TTL. Yeah we should research and document upsides/downsides of this at some point

Reply:
Alright, now that you mention it, I think it's probably fine to be a bit more conservative here. I've updated it to 2 weeks. Wdyt?